CMakeLists.txt

```diff
@@ -31,12 +31,26 @@ if(WITH_TRACE)
     add_definitions("-DWITH_TRACE")
 endif()
 
-# TBB support
-set(WITH_TBB OFF CACHE BOOL "use TBB for on-node threading" )
-if(WITH_TBB)
+#threading model selection
+if(THREADING_MODEL MATCHES "tbb")
+    # TBB support
+    set(WITH_TBB OFF CACHE BOOL "use TBB for on-node threading" )
```

Comment:
the `WITH_TBB` and `WITH_OMP` variables below are no longer need if we use `THREADING_MODEL` to describe the threading backend.